File: bindings/node/cellforge.js
const ffi = require('ffi-napi');
const path = require('path');
const fs = require('fs');

class CellForge {
    constructor(libPath = null) {
        // Default to ../../lib/libcellforge.so
        const basePath = path.join(__dirname, '..', '..', 'lib', 'libcellforge.so');
        const finalPath = libPath || basePath;
        
        if (!fs.existsSync(finalPath)) {
            throw new Error(`Missing CellForge library at ${finalPath}\nRun \`make\` from the project root to build it.`);
        }
        
        // Load the shared library
        this._lib = ffi.Library(finalPath, {
            'cellforge': ['string', []]
        });
    }
    
    hello() {
        return this._lib.cellforge();
    }
}

module.exports = CellForge; 

File: bindings/ruby/cellforge.rb
require 'fiddle'
require 'fiddle/import'

class CellForge
  def initialize(lib_path = nil)
    # Default to ../../lib/libcellforge.so
    base_path = File.join(File.dirname(__FILE__), '..', '..', 'lib', 'libcellforge.so')
    final_path = lib_path || base_path
    
    unless File.exist?(final_path)
      raise "Missing CellForge library at #{final_path}\nRun `make` from the project root to build it."
    end
    
    # Load the shared library
    @lib = Fiddle.dlopen(final_path)
    
    # Get the cellforge function
    @cellforge_func = Fiddle::Function.new(
      @lib['cellforge'],
      [],
      Fiddle::TYPE_VOIDP
    )
  end
  
  def hello
    result_ptr = @cellforge_func.call
    result_ptr.to_s
  end
end 